cmd/neofs-cli/modules/tree/get_by_path.go

```diff
@@ -0,0 +1,102 @@
+package tree
+
+import (
+	"crypto/sha256"
+	"errors"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/common"
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/commonflags"
+	"github.com/nspcc-dev/neofs-node/cmd/neofs-cli/internal/key"
+	"github.com/nspcc-dev/neofs-node/pkg/services/tree"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	"github.com/spf13/cobra"
+)
+
+var getByPathCmd = &cobra.Command{
+	Use:   "get-by-path",
+	Short: "Get a node by its path",
+	Run:   getByPath,
+	PersistentPreRun: func(cmd *cobra.Command, _ []string) {
+		commonflags.Bind(cmd)
+	},
+}
+
+func initGetByPathCmd() {
+	commonflags.Init(getByPathCmd)
+	initCTID(getByPathCmd)
+
+	ff := getByPathCmd.Flags()
+
+	// tree service does not allow any attribute except
+	// the 'FileName' but that's a limitation of the
+	// current implementation, not the rule
+	//ff.String(pathAttributeFlagKey, "", "Path attribute")
+	ff.String(pathFlagKey, "", "Path to a node")
+
+	ff.Bool(latestOnlyFlagKey, false, "Look only for the latest version of a node")
+
+	_ = cobra.MarkFlagRequired(ff, commonflags.RPC)
+}
+
+func getByPath(cmd *cobra.Command, _ []string) {
+	pk := key.GetOrGenerate(cmd)
+
+	cidRaw, _ := cmd.Flags().GetString(cidFlagKey)
+
+	var cnr cid.ID
+	err := cnr.DecodeString(cidRaw)
+	common.ExitOnErr(cmd, "decode container ID string: %w", err)
+
+	tid, _ := cmd.Flags().GetString(tidFlagKey)
+	if tid == "" {
+		common.ExitOnErr(cmd, "", errors.New("empty tree ID"))
+	}
+
+	ctx := cmd.Context()
+
+	cli, err := _client(ctx)
+	common.ExitOnErr(cmd, "client: %w", err)
+
+	rawCID := make([]byte, sha256.Size)
+	cnr.Encode(rawCID)
+
+	latestOnly, _ := cmd.Flags().GetBool(latestOnlyFlagKey)
+	path, _ := cmd.Flags().GetString(pathFlagKey)
+	//pAttr, _ := cmd.Flags().GetString(pathAttributeFlagKey)
+
+	req := new(tree.GetNodeByPathRequest)
+	req.Body = &tree.GetNodeByPathRequest_Body{
+		ContainerId:   rawCID,
+		TreeId:        tid,
+		PathAttribute: object.AttributeFileName,
+		//PathAttribute: pAttr,
+		Path:          strings.Split(path, "/"),
+		LatestOnly:    latestOnly,
+		AllAttributes: true,
+		BearerToken:   nil, // TODO: add token handling
+	}
+
+	common.ExitOnErr(cmd, "message signing: %w", tree.SignMessage(req, pk))
+
+	resp, err := cli.GetNodeByPath(ctx, req)
+	common.ExitOnErr(cmd, "rpc call: %w", err)
+
+	nn := resp.GetBody().GetNodes()
+	if len(nn) == 0 {
+		common.ExitOnErr(cmd, "", errors.New("node not found"))
+	}
```

Comment:
I don't think it is an error, it is a valid response.

Reply:
do you mean exit with `0` not with the `1`?

Reply:
Yes, an we can print nothing in this case. Maybe print `node not found` in verbose mode?